### PR TITLE
feat: propagate task priorities to resource pools [DET-4510]

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
This PR supports propagating task configured priorities to the resource pools.

## Test Plan
Tested manually & added unit test coverage.

<s>Note: Only the last two commits are related to this PR.</s>

### CODE REVIEW

File: master/internal/resourcemanagers/resource_pool.go [hunk 1]
```diff
@@ -151,6 +151,14 @@ func (rp *ResourcePool) getOrCreateGroup(
 		return g
 	}
 	g := &group{handler: handler, weight: 1}
+
+	if rp.config.Scheduler.Priority != nil {
+		if rp.config.Scheduler.Priority.DefaultPriority == nil {
+			panic("default priority is not configured")
```
Comment (shiyuann) [inline master/internal/resourcemanagers/resource_pool.go:157]: q: I thought we settled on a constant value for default priority. why we need to handle the nil case?

**Review by shiyuann — APPROVED** [review]
good work!

File: master/internal/resourcemanagers/resource_pool.go (hunk 1 above)
Comment (aaron276h) [reply master/internal/resourcemanagers/resource_pool.go:157]: It isn't configured when using fair_share, and this is just catching if we break that configuration logic somehow down the road. Can also just get rid of this.

Comment (shiyuann) [reply master/internal/resourcemanagers/resource_pool.go:157]: it's fine to keep it

**Pull request merged**